pkg/archive/archive.go

```diff
@@ -337,8 +328,11 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		return fmt.Errorf("Unhandled tar header type %d\n", hdr.Typeflag)
 	}
 
-	if err := os.Lchown(path, hdr.Uid, hdr.Gid); err != nil && Lchown {
-		return err
+	// Lchown is not supported on Windows
+	if runtime.GOOS != "windows" {
```

Comment:
The reason I'm not keen on factoring this out that was is that the pattern with other calls would be in system.LChown to return ErrNotSupportedPlatform for Windows, which isn't what we want at all. Having it return nil in the Windows implementation would be misleading. Hence I'd rather keep it like this. There are several other examples of runtime.GOOS checks in non- _windows/_Linux files in the existing code base.